pacman-helper.sh

```diff
@@ -545,9 +178,30 @@ quick_add () { # <file>...
 	test $# -gt 0 ||
 	die "Need at least one file"
 
-	# Create a temporary directory to work with
+	if test -z "$PACMANDRYRUN$azure_blobs_token"
+	then
+		azure_blobs_token="$(cat "$HOME"/.azure-blobs-token)" &&
+		test -n "$azure_blobs_token" ||
+		die "Could not read token from ~/.azure-blobs-token"
+	fi
+
+	if test -z "$PACMANDRYRUN$GITHUB_TOKEN"
+	then
+		die 'Need `GITHUB_TOKEN` to upload the files to `git-for-windows/pacman-repo`'
+	fi
+
+	# Create a shallow, sparse & partial clone of
+	# git-for-windows/pacman-repo to work with
 	dir="$(mktemp -d)" &&
-	mkdir "$dir/x86_64" "$dir/aarch64" "$dir/i686" "$dir/sources" ||
+	git -C "$dir" init &&
+	git -C "$dir" remote add origin https://github.com/git-for-windows/pacman-repo &&
+	git -C "$dir" config set remote.origin.promisor true &&
+	git -C "$dir" config set remote.origin.partialCloneFilter blob:none &&
+	git -C "$dir" config set core.sparseCheckout true &&
+	git -C "$dir" config set core.sparseCheckoutCone false &&
+	printf '%s\n' 'git-*.db*' 'git-*.files*' >"$dir"/.git/info/sparse-checkout &&
+	printf '%s\n' '/git-for-windows.db*' 'git-for-windows.files*' >"$dir"/.git/info/exclude &&
```

Comment:
I don't know the syntax here, but it seems odd there's a leading slash on just this one item.

Reply:
The resulting `exclude` file would be:

```text
/git-for-windows.db*
git-for-windows.files*
```

..which means, exclude _only_ files matching `git-for-windows.db*` from the root of the repository `/`, and exclude files matching `git-for-windows.files*` at any depth path.